fenics_ice/model.py

```diff
@@ -148,7 +148,17 @@ def alpha_from_data(self):
     def bglen_from_data(self):
         """Get bglen field from initial input data"""
         self.bglen = self.input_data.interpolate("Bglen", self.Q)
-
+        """Get bglen mask field from input data"""
+        self.bglen_mask = self.input_data.interpolate("Bglenmask", self.M)
+
+        """bglen_mask is currently just bglen where data available and nan elsehere"""
+        """Following is hacky way to convert this to 0 where nan, 1 elsewhere"""
+        temp_vec = self.bglen_mask.vector()[:]
```

Comment:
`self.bglen_mask.vector().get_local()` here, or can use the tlm_adjoint interface, `temp_vec = function_get_local(self.bglen_mask)`

Reply:
@jrmaddison  see new changes to this function